### ISSUE TITLE
Error/warning file location format not compliant with FXC nor Visual Studio

### ISSUE THREAD

**Issue by wbahnassi-if** [microsoft/DirectXShaderCompiler#2705]
The way DXC reports errors/warnings location is different than FXC and doesn't play nice with Visual Studio's Output window. Here is how each compiler is reporting error locations:

```
FXC:   Shader.hlsl(17,1-11): error X3000: blah blah blah
DXC:   Shader.hlsl:124:16: error: blah blah blah
```

In Visual Studio's Output window, double-clicking on the FXC location (or pressing F8) takes you immediately to the correct file and line. But doing so with the DXC format only opens the file but keeps the cursor at the start of the file as VS doesn't recognize the DXC way of specifying location within the file (no parenthesis).

Suggestion is to either revert to the FXC standard or expose a switch to allow us to specify which format to use for locating errors within files.

**Comment by gracejennings** [comment]
@wbahnassi-if Thanks for the report! I am not able to reproduce this issue with the latest version of the compiler and Visual Studio 2022.  Can you provide more information on your Visual Studio configuration and compiler version?

**Comment by wbahnassi-if** [comment]
Thanks @gracejennings for checking this out. The latest DXC release fixed the error/warning output format back to the standard way like old FXC did, and now Visual Studio can navigate to the correct location. Here is an error log from DXC now:

Shader.hlsl(53,85-89): error X3000: unrecognized identifier 'zxzxs'

The line above came from using Visual Studio 2019 Community (16.11.13) with DXC invoked through the DXCompiler.dll from Windows 10 SDK (10.0.22000.0).

I'll close this issue as fixed. Thanks!

**Comment by neversleeping** [comment]
Looks like this issue actually never went away. Visual Studio really outputs errors in format with braces. The problem is that I found no explanation, how. DXC itself outputs a format "file:line:column" and looks like Visual Studio makes some internal magic to output error differently. But when you want to use DXC directly and then push it's output into Visual Studio (for example, in some custom extension), output have format that Visual Studio can't interpret. I think that we need a compilation flag to switch output format.

**Comment by alecazam** [comment]
VSCode I believe parses the file:line: error: format, but VS does not.   glslc also uses the same format.  It would really help if VS would update their parser to handle both warning/error cases.  Otherwise, we have to update all our tools to file(line): error: style to get clickthrough to errors.

For example, here's shaderc saying VS should fix their parser.  Even though it's a few line change.
https://github.com/google/shaderc/issues/1472
